python/cudf/cudf/core/_internals/where.py

```diff
@@ -234,9 +236,16 @@ def where(
 
     if isinstance(frame, DataFrame):
         if hasattr(cond, "__cuda_array_interface__"):
-            cond = DataFrame(
-                cond, columns=frame._column_names, index=frame.index
-            )
+            if isinstance(cond, DataFrame):
+                cond = DataFrame(
+                    cond, columns=frame._column_names, index=frame.index
+                )
+            else:
+                cond = DataFrame(
+                    {name: cond for name in frame.columns},
+                    columns=frame._column_names,
+                    index=frame.index,
```

Comment:
```suggestion
                    {name: cond for name in frame.column_names},
                    index=frame.index,
```

Because `DataFrame.columns` is a property that is evaluated dynamically, `frame._column_names` is much more efficient than `frame.columns`. Secondly, we don't need `columns=` when constructing a DataFrame from a dictionary. The column names are inferred from the keys of the dictionary.

Reply:
thanks! updated to `frame._column_names` and removed `columns=`